tools/lsp/preview.rs

```diff
@@ -960,17 +960,20 @@ fn finish_parsing(preview_url: &Url, previewed_component: Option<String>) {
             }
         }
 
-        let uses_widgets = document_cache
-            .get_document(preview_url)
-            .and_then(|d| d.node.as_ref())
-            .map(|n| {
-                n.ImportSpecifier().any(|is| {
+        let uses_widgets =
+            document_cache.get_document_with_dependencies(preview_url).iter().any(|d| {
+                let Some(node) = &d.node else {
+                    return false;
+                };
+
+                eprintln!("Looking at document {:?}", node.source_file.path());
+
+                node.ImportSpecifier().any(|is| {
```

Comment:
Instead of looking for import std-widgets.slint,  i guess it's easier to just check if std-widgets.slint is in the list of document.

In fact, you could terminate the recursive function once you find it, instead of having a helper function that builds a vector.

Reply:
Looking for the std-widgets does not work: We load them always, so that we can auto-import the widgets (both in the LSP and the live-preview part). I think I really need to do imports dance. 

I could probably do it more efficiently, but having a list of documents with all dependencies included seemed like something useful to have.

Reply:
I'm saying looking if std-widgets is in the list returned by `get_document_with_dependencies`

> I could probably do it more efficiently, but having a list of documents with all dependencies included seemed like something useful to have.

Be careful not falling for the [YAGNI trap](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Reply:
OK, ok, I'll change it.

We have way more special purpose code that covers exactly one use case and is not reusable at all. Well, I guess the special purpose code is fast...